third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -201,34 +210,44 @@ void StreamPipeliner::initSchedule(int maxIndirectionLevel) {
   int lastStage = numStages - 1;
   config.resize(SCHED_TAIL + 1);
 
-  bool isMultibuf = numStages > (2 + maxIndirectionLevel);
-  if (prefetch) {
-    // Prefetch Schema cluster order and staging.
+  // Calculate the number of buffers needed for each load.
+  // TODO: Use the precise number of buffers needed by the particular load.
+  numBuffers =
+      numStages - globalPrefetch - localPrefetch - maxIndirectionLevel - 1;
+  LDBG("deduced max shared memory buffer number = " << numBuffers);
+
+  int localPrefetchStage = lastStage - localPrefetch;
+  if (numBuffers > 1) {
+    // Streaming Schema cluster order and staging for multi-buffer.
     // for i in (...):
     //   local_stores: stage=i+1
     //   global_loads: stage=i+2
+    //   local_load:   stage=i
     //   compute:      stage=i
-    //   local_load:   stage=i+1
     //   tail:         stage=i
-    config[SCHED_LOCAL_STORE] = {lastStage - 1, schedule.clusters.newAtBack()};
+    int localStoreStage = localPrefetchStage - (numBuffers - 1);
+    config[SCHED_LOCAL_STORE] = {localStoreStage,
+                                 schedule.clusters.newAtBack()};
     auto cluster1 = schedule.clusters.newAtBack();
     config[SCHED_GLOBAL_LOAD] = {0, cluster1};
+    config[SCHED_LOCAL_LOAD] = {localPrefetchStage, cluster1};
     config[SCHED_COMPUTE] = {lastStage, cluster1};
-    config[SCHED_LOCAL_LOAD] = {lastStage - 1, schedule.clusters.newAtBack()};
     config[SCHED_TAIL] = {lastStage, schedule.clusters.newAtBack()};
-  } else if (isMultibuf) {
-    // Streaming Schema cluster order and staging for multi-buffer.
+  } else if (localPrefetch) {
+    // Local Prefetch Schema cluster order and staging.
     // for i in (...):
     //   local_stores: stage=i+1
     //   global_loads: stage=i+2
-    //   local_load:   stage=i
     //   compute:      stage=i
+    //   local_load:   stage=i+1
```

Comment:
Similarly here--not sure these stages are still properly reflecting the changes.